engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/TimerEntity.java

```diff
@@ -100,6 +110,28 @@ protected void preExecute(CommandContext commandContext) {
     }
   }
 
+  protected boolean isReevaluateTimeCycleWhenDue(CommandContext commandContext) {
+    return commandContext.getProcessEngineConfiguration().isReevaluateTimeCycleWhenDue();
+  }
+
+  protected boolean isCycleExpression() {
+    String jobConfiguration = jobDefinition.getJobConfiguration();
+    return jobConfiguration.contains("CYCLE: #") || jobConfiguration.contains("CYCLE: $");
+  }
+
+  protected String adjustRepeatBasedOnNewExpression(String expressionValue) {
+    String changedRepeat = expressionValue; // changed to a cron expression
+    if (expressionValue.startsWith("R")) { // changed to a repeatable interval
+      if (repeat.startsWith("R")) { // was repeatable interval
+        changedRepeat = expressionValue.replace("/", "/" + repeat.split("/")[1] + "/");
+      } else {// was a cron expression
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+        changedRepeat = expressionValue.replace("/", "/" + sdf.format(ClockUtil.getCurrentTime()) + "/");
+      }
+    }
+    return changedRepeat;
+  }
```

Comment:
❓ Do we initially use something like this as well when parsing the expression from the model? If so, could we rather reuse it so we are aligned in how we parse an expression into a repeat value so they don't diverge in the future?

Reply:
* Resolving the `dueDate` is done here: https://github.com/camunda/camunda-bpm-platform/blob/e573859c094bff126b86cffc2d0295068a7d8ad1/engine/src/main/java/org/camunda/bpm/engine/impl/calendar/CycleBusinessCalendar.java#L44-L60
However, we need to change the whole repeat so in this sense they are not comparable, and can't reuse much.

* The initial resolving/calculation of the `repeat` is done here: https://github.com/camunda/camunda-bpm-platform/blob/e573859c094bff126b86cffc2d0295068a7d8ad1/engine/src/main/java/org/camunda/bpm/engine/impl/jobexecutor/TimerDeclarationImpl.java#L98-L109
However, the initialization of the field differs of the recalculation here.

I will check again if anything can be reused.

Reply:
As the logic is quite specific for this feature requirements and use cases, I would avoid reusing the existing implementation. I am still open for suggestions.